diff-containers/src/Data/Map/Diff/Strict/Internal.hs

```diff
@@ -212,6 +216,20 @@ null (Diff m) = Map.null m
 size :: Diff k v -> Int
 size (Diff m) = Map.size m
 
+-- | @'sizeEffect' d@ returns the number of inserts minus the number of deletes
+-- for a diff @d@.
+--
+-- Applying @d@ to a 'Map' should increase the size of the 'Map' by this number,
+-- but only if @d@ does not contain redundant deletes with respect to the 'Map':
+-- a delete of a key that does not exist in the 'Map'.
```

Comment:
What if one inserts the same key twice? Could we add some examples to illustrate this?

Reply:
Also a good point! Not only applying diffs, but also how we `mappend` them has some preconditions for `sizeEffect` to provide useful information. In the UTxO case however, we're satisfying these preconditions